logging/sinks.js

```diff
@@ -0,0 +1,217 @@
+ * @param {string} name The name of the sink to update.
+ * @param {object} metadata The new metadata for the sink.
+ * @param {function} callback The callback function.
+ */
+function updateSink (options, callback) {
+  var sink = logging.sink(options.name);
+
+  // See https://googlecloudplatform.github.io/gcloud-node/#/docs/logging/latest/logging/sink
+  sink.setMetadata(options.metadata, function (err) {
+    if (err) {
+      return callback(err);
+    }
+
+    console.log('Updated sink: %s', options.name);
+    return callback(null);
+  });
+}
+// [END update_sink]
+
+// [START delete_sink]
+/**
+ * Delete a sink.
+ *
+ * @param {string} name The name of the sink to delete.
+ * @param {function} callback The callback function.
+ */
+function deleteSink (name, callback) {
+  var sink = logging.sink(name);
+
+  // See https://googlecloudplatform.github.io/gcloud-node/#/docs/logging/latest/logging/sink
+  sink.delete(function (err) {
+    if (err) {
+      return callback(err);
+    }
+
+    console.log('Deleted sink: %s', name);
+    callback(null);
+  });
+}
+// [END delete_sink]
+// [END all]
+
+// The command-line program
+var cli = require('yargs');
+
+var program = module.exports = {
+  createSink: createSink,
+  getSinkMetadata: getSinkMetadata,
+  listSinks: listSinks,
+  updateSink: updateSink,
+  deleteSink: deleteSink,
+  main: function (args) {
+    // Run the command-line program
+    cli.help().strict().parse(args).argv;
+  }
+};
+
+cli
+  .demand(1)
+  .command('create <name> <destination>', 'Create a new sink with the given name and destination.', {
+    filter: {
+      alias: 'f',
+      type: 'string',
+      requiresArg: true,
+      description: 'Optional. Only log entries matching the filter are written.'
+    },
+    type: {
+      alias: 't',
+      demand: true,
+      type: 'string',
+      choices: ['bucket', 'dataset', 'topic'],
+      requiresArg: true,
+      description: 'The type of destination.'
+    }
+  }, function (options) {
+    program.createSink(options, console.log);
+  })
+  .command('get <name>', 'Get the metadata for the specified sink.', {}, function (options) {
+    program.getSinkMetadata(options.name, console.log);
+  })
+  .command('list', 'List all sinks in the authenticated project.', {}, function () {
+    program.listSinks(console.log);
+  })
+  .command('update <name> <metadata>', 'Update the metadata for the specified sink.', {}, function (options) {
+    try {
+      options.metadata = JSON.parse(options.metadata);
+      program.updateSink(options, console.log);
+    } catch (err) {
+      console.error('"metadata" must be a valid JSON string!');
+    }
+  })
+  .command('delete <name>', 'Delete the specified sink.', {}, function (options) {
+    program.deleteSink(options.name, console.log);
+  })
+  .example('node $0 create my-sink my-bucket --type bucket', 'Create a new sink named "my-sink" that exports logs to a Cloud Storage bucket.')
```

Comment:
Can we infer `type` from the arguments, *or* do something like `--bucket my-bucket`/`--topic my-topic`?

That would save us a command-line argument...